scripts/backup/backup

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+UMBREL_ROOT="$(readlink -f $(dirname "${BASH_SOURCE[0]}")/../..)"
+BACKUP_FOLDER="backup"
+BACKUP_ROOT="${UMBREL_ROOT}/${BACKUP_FOLDER}"
+BACKUP_FILE="${UMBREL_ROOT}/backup.tar.gz.pgp"
+
+check_dependencies () {
+  for cmd in "$@"; do
+    if ! command -v "$cmd" >/dev/null 2>&1; then
+      echo "This script requires \"${cmd}\" to be installed"
+      exit 1
+    fi
+  done
+}
+
+check_dependencies openssl tar gpg shuf curl
+
+# Deterministically derives 128 bits of cryptographically secure entropy
+derive_entropy () {
+  identifier="${1}"
+  umbrel_seed=$(cat "${UMBREL_ROOT}/db/umbrel-seed/seed") || true
+
+  # TODO: Check this correctly handles when seed file doesn't exist!
+  if [[ -z "$umbrel_seed" ]] || [[ -z "$identifier" ]]; then
+    echo "Missing derivation parameter, this is unsafe, exiting."
+    rm -f "${UMBREL_ROOT}/statuses/backup-in-progress"
+    exit 1
+  fi
+
+  # We need `sed 's/^.* //'` to trim the "(stdin)= " prefix from some versions of openssl
+  echo -n "${identifier}" | openssl dgst -sha256 -hmac "${umbrel_seed}" | sed 's/^.* //'
+}
+
+# Make sure an update is not in progres
+if [[ -f "${UMBREL_ROOT}/statuses/backup-in-progress" ]]; then
+    echo "A backup is already in progress. Exiting now."
+    exit 1
+fi
+
+echo "Creating lock..."
+touch "${UMBREL_ROOT}/statuses/backup-in-progress"
+
+[[ -f "${UMBREL_ROOT}/.env" ]] && source "${UMBREL_ROOT}/.env"
+BITCOIN_NETWORK=${BITCOIN_NETWORK:-mainnet}
+
+[[ -d "${BACKUP_ROOT}" ]] && rm -rf "${BACKUP_ROOT}"
+
+echo "Deriving keys..."
+
+backup_id=$(derive_entropy "umbrel_backup_id")
+encryption_key=$(derive_entropy "umbrel_backup_encryption_key")
+
+echo "Creating backup..."
+
+mkdir -p "${BACKUP_ROOT}"
+
+cp --archive "${UMBREL_ROOT}/lnd/data/chain/bitcoin/${BITCOIN_NETWORK}/channel.backup" "${BACKUP_ROOT}/channel.backup"
```

Comment:
Why grab a file from underneath a running instance of `lnd`, and not ex:

```sh
lncli exportchanbackup --all | jq . | tar -czvf "${BACKUP_ROOT}/lnd-channels-backup.json.tgz"
```

Reply:
That's the file we're watching fro changes and it's explicitly designed to be safe to copy:

>On-Disk channel.backup
>There are multiple ways of obtaining SCBs from lnd. The most commonly used method will likely be via the channels.backup file that's stored on-disk alongside the rest of the chain data. This is a special file that contains SCB entries for all currently open channels. Each time a channel is opened or closed, this file is updated on disk in a safe manner (atomic file rename). As a result, unlike the channel.db file, it's always safe to copy this file for backup at ones desired location. 

https://github.com/lightningnetwork/lnd/blob/master/docs/recovery.md#on-disk-channelbackup

Reply:
TIL! thank you